content/6_Advanced/Persistent.mdx

```diff
@@ -127,13 +141,34 @@ int main() {
 
 </LanguageSection>
 
-### Problems
+### Application 1 - Static 2D Range Sums on Large Grids
 
-A nice thing about persistent segment trees is that they can be used for online
-2D static range sum queries in $\mathcal{O}(\log N)$ time (think about it like
-prefix sums).
+Persistent segment trees can be used for online 2D static range sum queries in
+$\mathcal O(\log N)$ time (think of it like prefix sums.)
 
-<IncompleteSection />
+Note that 2D Fenwick trees with coordinate compression often also work for this
+(and are easier to implement), but it's still good to know this application.
+
+### Application 2 - Largest Interval Completely Inside a Range
+
+Consider the following problem:
+
+> Given $N$ intervals on the number line, answer $Q$ queries of the form "what
+> is the largest interval completely contained inside the range $[x, y]$?
+>
+> $N, Q \leq 10^5$.
+
+Since each interval has two dimensions (i.e. left and right endpoints $l_i$ and
+$r_i$), we can view it as a *point* on the number line at $l_i$ with "value"
+$r_i - l_i$ that was inserted at time $r_i$.
+
+Now, each query becomes "what is the most valuable point in the range $[x, y]$
+that was inserted at or before time $y$?" This is much easier to handle, so we
+can solve this problem in $\mathcal O(Q \log N)$ time.
```

Comment:
is there somewhere you can submit this?

Reply:
I don't think so. @frodakcin @HarryWangATX can you guys try to make this a problem on the cf group please?